### PR TITLE
test: add .tree for Allo.sol

### PR DESCRIPTION
**Pull request by 0xAustrian**
Closes DTT-281

### CODE REVIEW

File: contracts/core/Allo.sol [hunk 1]
```diff
@@ -716,7 +716,7 @@ contract Allo is
     }
 
     /// @dev Logic copied from ERC2771ContextUpgradeable OZ contracts
-    function _msgData() internal view virtual override returns (bytes calldata) {
```
Comment (0xAustrian) [inline contracts/core/Allo.sol:719]: I had to remove the `virtual` keyword so smock does not crash when generating the mock. Looks like the return paramater of type `calldata` is messing with the generation.

File: test/foundry/core/AlloUnit.tree [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+    └── it should call removePoolManagers
+
+Allo::recoverFunds
+├── when sender is not owner
+│   └── it should revert
+└── when sender is owner
+    ├── when token is native
+    │   └── it should transfer native token
+    └── when token is not native
+        └── it should transfer token
```
Comment (0xOneTony) [inline test/foundry/core/AlloUnit.tree:85]: for the tests let's make sure we also check it transfers the correct amount

File: package.json [hunk 3]
```diff
@@ -29,7 +29,8 @@
     "deploy-deployer": "npx hardhat run scripts/core/deployContractFactory.ts --network",
     "/////// deploy-test ///////": "echo 'deploy test scripts'",
     "create-profile": "npx hardhat run scripts/test/createProfile.ts --network",
-    "create-pool": "npx hardhat run scripts/test/createPool.ts --network"
+    "create-pool": "npx hardhat run scripts/test/createPool.ts --network",
+    "smock": "smock-foundry --contracts contracts/core"
```
Comment (0xOneTony) [inline package.json:33]: Should we specify the full path? we currently only want Allo.sol to get generated

File: contracts/core/Allo.sol [hunk 4]
```diff
@@ -464,14 +464,14 @@ contract Allo is
     /// @notice Internal function to check is caller is pool manager
     /// @param _poolId The pool id
     /// @param _address The address to check
-    function _checkOnlyPoolManager(uint256 _poolId, address _address) internal view {
+    function _checkOnlyPoolManager(uint256 _poolId, address _address) internal view virtual {
```
Comment (0xOneTony) [inline contracts/core/Allo.sol:467]: Why do we add virtual in all these calls? Is it needed for smock?

File: package.json (hunk 3 above)
Comment (0xAustrian) [reply package.json:33]: I tried but fails with `[Error: ENOTDIR: not a directory, scandir '/.../.../.../.../contracts/core/Allo.sol']`. I think smock only allows folders

File: contracts/core/Allo.sol (hunk 4 above)
Comment (0xAustrian) [reply contracts/core/Allo.sol:467]: [Yes!](https://github.com/defi-wonderland/smock-foundry?tab=readme-ov-file#gotchas)

File: test/foundry/core/AlloUnit.tree (hunk 2 above)
Comment (0xAustrian) [reply test/foundry/core/AlloUnit.tree:85]: Yes, here I'm assuming that we will add a `expectCall` with exact parameters.

**Review by thelostone-mc — APPROVED** [review]

**Review by 0xOneTony — APPROVED** [review]

**Pull request merged**